src/corrgi/estimators/estimator.py

```diff
@@ -0,0 +1,40 @@
+import abc
+from abc import abstractmethod
+from typing import Callable
+
+import numpy as np
+from gundam.gundam import tpcf, tpcf_wrp
+from lsdb import Catalog
+
+from corrgi.correlation.correlation import Correlation
+from corrgi.correlation.projected_correlation import ProjectedCorrelation
+
+
+class Estimator(abc.ABC):
+    """Estimator base class"""
+
+    def __init__(self, correlation: Correlation):
+        self.correlation = correlation
+
+    def compute_auto_estimate(self, catalog: Catalog, random: Catalog) -> np.ndarray:
+        """Computes the auto-correlation for this estimator"""
+        counts = self.compute_autocorrelation_counts(catalog, random)
+        num_galaxies = catalog.hc_structure.catalog_info.total_rows
+        num_random = random.hc_structure.catalog_info.total_rows
+        subroutine_args = self._get_auto_args(*counts, num_galaxies, num_random)
+        wth, _ = self._get_auto_subroutine()(*subroutine_args)
```

Comment:
What the hell?

Reply:
I pushed some changes to improve clarity!